spec/services/offered_course_option_details_check_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe OfferedCourseOptionDetailsCheck do
+  let(:course_option) { create(:course_option, study_mode: :full_time) }
+
+  it 'throws an InvalidProviderError when the provider does not correspond to the course option' do
```

Comment:
can you add a test for a valid course option?

Reply:
Done ✅